File: WEEK_4_C/Q10.c
#include <stdio.h>
#include <stdlib.h>

struct Node{ int data; struct Node* next; };

int main(){
    int n=7,k=3;
    struct Node *head=NULL,*last=NULL,*t;
    for(int i=1;i<=n;i++){
        t=(struct Node*)malloc(sizeof(struct Node));
        t->data=i;
        if(!head){ head=t; head->next=head; last=head; }
        else{ last->next=t; t->next=head; last=t; }
    }
    struct Node *p=head,*q=NULL;
    while(p->next!=p){
        for(int i=1;i<k;i++){ q=p; p=p->next; }
        q->next=p->next; free(p); p=q->next;
    }
    printf("Survivor=%d\n",p->data);
    return 0;
}

File: WEEK_3_Python/Q10.py
class Node:
    def __init__(self, data):
        self.data, self.next = data, None

def print_list(head):
    cur = head
    while cur:
        print(cur.data, end=" ")
        cur = cur.next
    print()

# Example
head = Node(1); head.next = Node(2); head.next.next = Node(3)
print_list(head)   
